unit_tests/library/route.cpp

```diff
@@ -50,6 +50,7 @@ BOOST_AUTO_TEST_CASE(test_route_same_coordinates_fixture)
           json::Array{{json::Object{
               {{"distance", 0.},
                {"duration", 0.},
+               {"weight_name", "duration"},
```

Comment:
Looks like we are not exposing the `weight` property for the `Route` and `RouteLeg` objects. This needs an adjustment in `json_factory.cpp` and `src/engine/guidance/assemble_route.cpp`, `include/engine/guidance/assemble_leg.hpp`